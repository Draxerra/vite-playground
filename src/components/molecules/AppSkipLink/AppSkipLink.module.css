@import "/src/css/abstracts/media-queries.css";

.app-skip-link {
    position: fixed;
    top: calc(var(--outline-width) + var(--outline-offset) + 10px);
    left: calc(var(--outline-width) + var(--outline-offset) + 10px);

    text-decoration: none;
    transform: translateY(-200%);

    @media (--enhanced-motion) {
        transition:
            transform
            calc(var(--transition-duration) * 2)
            var(--transition-timing-function);
    }

    &:focus-visible {
        transform: translateY(0);
    }
}
