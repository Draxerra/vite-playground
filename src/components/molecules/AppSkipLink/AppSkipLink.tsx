import React, { FC } from "react";
import classNames from "classnames";

import AppButton from "~/components/atoms/AppButton";

import styles from "./AppSkipLink.module.css";

type AppSkipLinkProps = JSX.IntrinsicElements["a"];
const AppSkipLink: FC<AppSkipLinkProps> = ({ className, ...props }) => (
  <AppButton
    {...props}
    className={classNames(className, styles.appSkipLink)}
    tag="a"
  >
    Skip to main content
  </AppButton>
);

export default AppSkipLink;
