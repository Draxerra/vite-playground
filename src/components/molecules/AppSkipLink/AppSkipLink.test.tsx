import React from "react";
import AppSkipLink from "./AppSkipLink";
import { render } from "@testing-library/react";

describe("App Skip Link", () => {
  it("should match snapshot", () => {
    const skipLink = render(<AppSkipLink href="#test" />).asFragment();
    expect(skipLink).toMatchSnapshot();
  });
});
