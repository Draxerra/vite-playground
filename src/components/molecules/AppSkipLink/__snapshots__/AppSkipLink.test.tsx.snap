// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Skip Link should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="appSkipLink appButton"
    href="#test"
  >
    Skip to main content
  </a>
</DocumentFragment>
`;
